kernel/init.c

```diff
@@ -433,10 +433,8 @@ static void bg_thread_main(void *unused1, void *unused2, void *unused3)
 #endif /* CONFIG_STACK_POINTER_RANDOM */
 	boot_banner();
 
-#if defined(CONFIG_CPP)
-	void z_cpp_init_static(void);
-	z_cpp_init_static();
-#endif /* CONFIG_CPP */
+	void z_init_static(void);
+	z_init_static();
```

Comment:
Let's pull this code into init.c so it can be in-lined into bg_thread_main instead of being a separate function. It's tiny, and merging should make the resulting code smaller.

Reply:
As commented here https://github.com/zephyrproject-rtos/zephyr/pull/74682#issuecomment-2185273088 , I think it is best to just leave it as is by now. This PR only moves code which already exists and enables it without CPP without optimizing it further.